app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/DraftExciseMovementController.scala

```diff
@@ -18,48 +18,61 @@ package uk.gov.hmrc.excisemovementcontrolsystemapi.controllers
+import uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.PushNotificationConnector
 import uk.gov.hmrc.excisemovementcontrolsystemapi.controllers.actions.{AuthAction, ParseXmlAction, ValidateErnInMessageAction}
-import uk.gov.hmrc.excisemovementcontrolsystemapi.models.ExciseMovementResponse
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.{ErrorResponse, ExciseMovementResponse}
 import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.ValidatedXmlRequest
 import uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages.{IE815Message, IEMessage}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.notification.Constants
 import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.Movement
 import uk.gov.hmrc.excisemovementcontrolsystemapi.services.{MovementService, SubmissionMessageService, WorkItemService}
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
+import java.time.Instant
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 import scala.xml.NodeSeq
 
 @Singleton
 class DraftExciseMovementController @Inject()(
-                                               authAction: AuthAction,
-                                               xmlParser: ParseXmlAction,
-                                               validateErnInMessageAction: ValidateErnInMessageAction,
-                                               movementMessageService: MovementService,
-                                               workItemService: WorkItemService,
-                                               submissionMessageService: SubmissionMessageService,
-                                               cc: ControllerComponents
-                                             )(implicit ec: ExecutionContext)
+  authAction: AuthAction,
+  xmlParser: ParseXmlAction,
+  validateErnInMessageAction: ValidateErnInMessageAction,
+  movementMessageService: MovementService,
+  workItemService: WorkItemService,
+  submissionMessageService: SubmissionMessageService,
+  notificationConnector: PushNotificationConnector,
+  cc: ControllerComponents)(implicit ec: ExecutionContext)
   extends BackendController(cc) {
 
   def submit: Action[NodeSeq] =
     (authAction andThen xmlParser andThen validateErnInMessageAction).async(parse.xml) {
       implicit request: ValidatedXmlRequest[NodeSeq] =>
 
-        submissionMessageService.submit(request).flatMap {
-          case Right(_) => handleSuccess
-          case Left(error) => Future.successful(error)
+        request.headers.get(Constants.XClientIdHeader) match {
+          case Some(clientId) =>
+            notificationConnector.getBoxId(clientId).flatMap {
+              case Right(box) => submitMessage(box.boxId)
+              case Left(error) => Future.successful(error)
+            }
+          case _ => Future.successful(BadRequest(Json.toJson(ErrorResponse(Instant.now, "ClientId error", "Request header is missing clientId"))))
```

Comment:
Is there a task to add this to the API spec? Or should it be done as part of this commit?

Reply:
Actually I was thinking about this today and I am not sure if there is a task for this. But yes I think it should be a task to be done with the notification story. I will check if you have a ticked linked to the push notification and I will add one if have none